### ISSUE TITLE
Upload Feature

### ISSUE THREAD

**Issue by systems-rebooter** [devsnd/cherrymusic#533]
Hey, 

I know that cherrymusic is positioning itself as RO software which makes no harm to collection, but upload feature to different location will be still super-useful. What do you think?

**Comment by systems-rebooter** [comment]
I definitely love how super-fast @6arms1leg tagging issues after creation! Its like in seconds... +1

**Comment by devsnd** [comment]
I agree that it would be crazy awesome to be able to upload. My only concern is the amount of possible vulnerabilities to which CM would be opened up if users could upload arbitrary data... Anyway, I'd love to have the feature as well, but there is so much left to do (all kinds of bugs that scream for rewrites of parts of CM).

I'll leave this issue open, maybe someone else wants to implement this?

**Comment by systems-rebooter** [comment]
Hey @devsnd 

I'm trying to make upload feature alive. Its long and hard way, since I'm a not programmer.. Thus, totally stucked with displaying upload form in left frame (in same / new window its showing up ok, trough.. which is wrong, since it will mess musing playing / navigation). 

![screen shot 2015-05-03 at 12 14 02](https://cloud.githubusercontent.com/assets/8888190/7444497/8518d2f6-f18e-11e4-8eca-df3b9abb0588.png)
 
What should I change and where to make this work, because right now  I even simply can't display dummy `<html>` page in left frame.. The biggest achievement i was up to is duplication of `showPlaylists()` / `displayMessageOfTheDay()` / `loadBrowser()` functions, followed by clicked `upload` link

Thanks much!

**Comment by systems-rebooter** [comment]
Found a modal workaround. 

![screen shot 2015-05-05 at 20 10 22](https://cloud.githubusercontent.com/assets/8888190/7478370/147b3b16-f363-11e4-8296-659ab3d19c9a.png)

**Comment by rajesh-battala** [comment]
Is Upload files feature patch is available  now?

**Comment by systems-rebooter** [comment]
I don't think so. Its not implemented because of security issues. To allow an end user to upload files to your website, is like opening another door for a malicious user to compromise your server (c) 

I added this feature by my own with server side PHP validation.

Good reads on upload security: 

https://www.acunetix.com/websitesecurity/upload-forms-threat/
http://stackoverflow.com/questions/256172/what-is-the-most-secure-method-for-uploading-a-file/18530956#18530956
